Public/GetQuestObjectiveIconForUnit.lua

```diff
@@ -0,0 +1,18 @@
+---@type QuestieNameplate
+local QuestieNameplate = QuestieLoader:ImportModule("QuestieNameplate")
+
+--- Returns the icon path for the quest objective marker for the given unit GUID.
+--- Only NPC GUIDs are valid inputs.
+---@param guid string
+---@return string | nil -- Path to the icon texture, or nil if no icon is available.
```

Comment:
Discussion of wanting to do string|nil or string? ;)
I doesn't actually matter so i think we can just mix them.

Reply:
I think for return types it's good to have them explicit, while for parameter's I prefer the `?` writing.